pyanaconda/modules/subscription/runtime.py

```diff
@@ -114,3 +158,110 @@ def run(self):
 
         # and finally set the dict to RHSM via the DBus API
         self._rhsm_config_proxy.SetAll(config_dict, "")
+
+
+class RegisterWithUsernamePasswordTask(Task):
+    """Register the system via username + password."""
+
+    def __init__(self, rhsm_register_server_proxy, username, password):
+        """Create a new registration task.
+
+        It is assumed the username and password have been
+        validated before this task has been started.
+
+        :param rhsm_register_server_proxy: DBus proxy for the RHSM RegisterServer object
+        :param str username: Red Hat account username
+        :param str password: Red Hat account password
+        """
+        super().__init__()
+        self._rhsm_register_server_proxy = rhsm_register_server_proxy
+        self._username = username
+        self._password = password
+
+    @property
+    def name(self):
+        return "Register with Red Hat account username and password"
+
+    def run(self):
+        """Register the system with Red Hat account username and password.
+
+        :return: True on success, raises RegistrationError on failure
+        :rtype: bool
+        :raises: RegistrationError
+        """
+        log.debug("subscription: registering with username and password")
+        with RHSMPrivateBus(self._rhsm_register_server_proxy) as private_bus:
+            try:
+                locale = os.environ.get("LANG", "")
+                private_register_proxy = private_bus.get_proxy(RHSM.service_name,
+                                                               RHSM_REGISTER.object_path)
+                # We do not yet support setting organization for username & password
+                # registration, so organization is blank for now.
+                organization = ""
+                private_register_proxy.Register(organization,
+                                                self._username,
+                                                self._password,
+                                                {},
+                                                {},
+                                                locale)
+                log.debug("subscription: registered with username and password")
+            except DBusError as e:
+                log.debug("subscription: failed to register with username and password: %s",
+                          str(e))
+                # RHSM exception contain details as JSON due to DBus exception handling limitations
+                exception_dict = json.loads(str(e))
+                # return a generic error message in case the RHSM provided error message is missing
+                message = exception_dict.get("message", _("Registration failed."))
+                raise RegistrationError(message)
```

Comment:
Please use `raise RegistrationError(message) from None`.

Reply:
Interesting, haven't known this. Let's do it. :)